.github/workflows/build-and-test.yml

```diff
@@ -25,23 +25,36 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - name: ubuntu-20.04 - clang
+          - name: ubuntu-20.04 - clang 10
             os: ubuntu-20.04
-            cc_compiler: clang
-            cxx_compiler: clang++
-          - name: ubuntu-22.04 - clang
+            cc_compiler: clang-10
+            cxx_compiler: clang++-10
+          - name: ubuntu-22.04 - clang 13
             os: ubuntu-22.04
-            cc_compiler: clang
-            cxx_compiler: clang++
-
-          - name: ubuntu-20.04 - gcc
+            cc_compiler: clang-13
+            cxx_compiler: clang++-13
+          - name: ubuntu-24.04 - clang 16
+            os: ubuntu-24.04
+            cc_compiler: clang-16
+            cxx_compiler: clang++-16
+
+          - name: ubuntu-20.04 - gcc 10
             os: ubuntu-20.04
-            cc_compiler: gcc
-            cxx_compiler: g++
-          - name: ubuntu-22.04 - gcc
+            cc_compiler: gcc-10
+            cxx_compiler: g++-10
+          - name: ubuntu-22.04 - gcc 10
             os: ubuntu-22.04
-            cc_compiler: gcc
-            cxx_compiler: g++
+            cc_compiler: gcc-10
+            cxx_compiler: g++-10
+          - name: ubuntu-24.04 - gcc 12
+            os: ubuntu-24.04
+            cc_compiler: gcc-12
+            cxx_compiler: g++-12
+
+          - name: macos-14 - clang
+            os: macos-14
+            cc_compiler: clang
+            cxx_compiler: clang++
```

Comment:
Also tested macos-12 and macos-13:

* macos-12 couldn't find python packages on pip... not accepting the universal binaries?): https://github.com/ScottTodd/iree-template-runtime-cmake/actions/runs/10204773640/job/28234045195
* macos-13 got Bad CPU type in executable (used universal binaries incorrectly?): https://github.com/ScottTodd/iree-template-runtime-cmake/actions/runs/10204853675/job/28234292155#step:12:30

Proceeding with what works out of the box for now.